activerecord/test/cases/connection_pool_test.rb

```diff
@@ -5,11 +5,15 @@
 
 module ActiveRecord
   module ConnectionAdapters
-    class ConnectionPoolTest < ActiveRecord::TestCase
+    module ConnectionPoolTests
```

Comment:
I reworked this file so that there are two ActiveRecord::TestCase subclasses that include these shared tests as a module with slightly different test setup and some overrides as to how threads (or fibers) are spun up, joined, `pass`'d to. This approach achieves decent test thoroughness while keeping the diff small, and it highlights a few things we might want to come back to, such as testing this more thoroughly in the Future with a Ruby 3.1+ Fiber Scheduler.